src/lib/endpoints.ts

```diff
@@ -1,39 +1,43 @@
-import { CHAIN_ID } from '@/lib/constants'
+import { CHAIN_ID, EVENTS_FROM_BLOCK } from '@/lib/constants'
+
+const CHAIN_ID_PARAM = `chainId=${CHAIN_ID}`
+const FROM_BLOCK_PARAM = `fromBlock=${EVENTS_FROM_BLOCK}`
 
 export const fetchAddressTokensEndpoint =
-  process.env.NEXT_PUBLIC_API_RWS_TOKEN_BY_ADDRESS || '/address/{{address}}/tokens?chainId={{chainId}}'
+  process.env.NEXT_PUBLIC_API_RWS_TOKEN_BY_ADDRESS ||
+  `/address/{{address}}/tokens?chainId={{chainId}}&${FROM_BLOCK_PARAM}`
 
 export const fetchPricesEndpoint =
   process.env.NEXT_PUBLIC_API_RWS_PRICES_BY_ADDRESS ||
-  `/price?addresses={{addresses}}&convert={{convert}}&chainId=${CHAIN_ID}`
+  `/price?addresses={{addresses}}&convert={{convert}}&${CHAIN_ID_PARAM}&${FROM_BLOCK_PARAM}`
 
 export const fetchNFTsOwnedByAddressAndNftAddress =
   process.env.NEXT_PUBLIC_API_RWS_NFT_BY_ADDRESS ||
-  `/address/{{address}}/nfts/{{nftAddress}}?chainId=${CHAIN_ID}`
+  `/address/{{address}}/nfts/{{nftAddress}}?${CHAIN_ID_PARAM}&${FROM_BLOCK_PARAM}`
 
 const PROPOSAL_CREATED_EVENT = '0x7d84a6263ae0d98d3329bd7b46bb4e8d6f98cd35a7adb45c274c8b7fd5ebd5e0'
 export const fetchProposalsCreatedByGovernorAddress =
   process.env.NEXT_PUBLIC_API_RWS_EVENTS_PROPOSALS_BY_ADDRESS ||
-  `/address/{{address}}/eventsByTopic0?topic0=${PROPOSAL_CREATED_EVENT}&chainId=${CHAIN_ID}&fromBlock={{fromBlock}}`
+  `/address/{{address}}/eventsByTopic0?topic0=${PROPOSAL_CREATED_EVENT}&${CHAIN_ID_PARAM}&${FROM_BLOCK_PARAM}`
 
 export const getNftInfo =
-  process.env.NEXT_PUBLIC_API_RWS_NFT_INFO || `/nfts/{{nftAddress}}?chainId=${CHAIN_ID}`
+  process.env.NEXT_PUBLIC_API_RWS_NFT_INFO || `/nfts/{{nftAddress}}?${CHAIN_ID_PARAM}&${FROM_BLOCK_PARAM}`
 
-export const getTokenHoldersOfAddress = `/address/{{address}}/holders?chainId=${CHAIN_ID}`
+export const getTokenHoldersOfAddress = `/address/{{address}}/holders?${CHAIN_ID_PARAM}&${FROM_BLOCK_PARAM}`
 
-export const getNftHolders = `/nfts/{{address}}/holders?chainId=${CHAIN_ID}`
+export const getNftHolders = `/nfts/{{address}}/holders?${CHAIN_ID_PARAM}&${FROM_BLOCK_PARAM}`
```

Comment:
These events are managed by the DAO, so I don't think we should use the CR related block. I think we could just leave them as they are.